File: 백준/Silver/10845. 큐/큐.java
import java.io.*;
import java.util.*;


public class Main {
    public void solution() throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st;
        
        Deque<Integer> dq = new LinkedList<>();
        int N = Integer.parseInt(br.readLine());
        for (int i=0;i<N;i++){
            st = new StringTokenizer(br.readLine());
            String q = st.nextToken();
            
            int state;
            switch (q){
                case "push":
                    int num = Integer.parseInt(st.nextToken());
                    dq.addLast(num);
                    break;
                case "pop":
                    state = dq.isEmpty() ? -1 : dq.pollFirst();
                    bw.write(state + "\n");
                    break;
                case "size":
                    state = dq.size();
                    bw.write(state + "\n");
                    break;
                case "empty":
                    state = dq.isEmpty() ? 1 : 0;
                    bw.write(state + "\n");
                    break;
                case "front":
                    state = dq.isEmpty() ? -1 : dq.peekFirst();
                    bw.write(state + "\n");
                    break;
                case "back":
                    state = dq.isEmpty() ? -1 : dq.peekLast();
                    bw.write(state + "\n");
                    break;
            }
        }
        
        br.close();
        bw.flush();
        bw.close();
    }
    
    public static void main(String[] args) throws Exception {
        new Main().solution();
    }
}

File: 백준/Silver/10845. 큐/큐.py
import sys
from collections import deque

input = sys.stdin.readline

N = int(input())

queue = deque()
for _ in range(N):
    query = input().strip().split()

    if query[0] == "push":
        queue.append(query[1])

    elif query[0] == "pop":
        if not queue:
            print(-1)

        else:
            print(queue.popleft())

    elif query[0] == "size":
        print(len(queue))

    elif query[0] == "empty":
        print(1 if not queue else 0)

    elif query[0] == "front":
        if not queue:
            print(-1)

        else:
            print(queue[0])

    else:
        if not queue:
            print(-1)

        else:
            print(queue[-1])